apps/src/p5lab/spritelab/CoreLibrary.js

```diff
@@ -107,6 +111,39 @@ export default class CoreLibrary {
     });
   }
 
+  drawVariableBubbles() {
+    this.variableBubbles.forEach(variable => {
+      const {name, location} = variable;
+      if (!name.length || !location) {
+        return;
+      }
+
+      const value = this.getVariableValue(name);
+      const text = `${name}: ${value}`;
+
+      // TODO: Confirm this handles shadow block locations appropriately
+      drawUtils.variableBubble(this.p5, location.x, location.y, text);
+    });
+  }
+
+  getVariableValue(variableName) {
+    if (!this.jsInterpreter) {
+      console.error('JS Interpreter not set in CoreLibrary');
+      return;
+    }
+
+    try {
+      // TODO: Check for existence of this.REFERENCE_ERROR and others in result,
+      // Or move this logic into the JSInterpreter and guarantee that it always returns a string
+      // TODO: Decide on what to display for undefined variables
+      const result = this.jsInterpreter.evaluateWatchExpression(variableName);
```

Comment:
I strongly feel it would be worth getting the Blockly variable name at this point so we can avoid reference errors. Instead of passing `variableName`, we'd basically want to pass `Blockly.JavaScript.getVariableName(variableName)` instead. I've confirmed locally that this works:

**Without suggested change**
![image](https://github.com/code-dot-org/code-dot-org/assets/43474485/d18a51bc-0b3a-4b38-b71f-131a73118da9)


**With change**
![image](https://github.com/code-dot-org/code-dot-org/assets/43474485/8b8e15c2-7d78-4e2f-83aa-0638bf025349)

Reply:
Yep, I got this working locally yesterday. I'll push it!